File: Java/Behavioral/Observer/Example/Show/src/ConcreteObserver.java
public class ConcreteObserver  implements Observer {
    private String name;
    private ConcreteSubject subject;

    public ConcreteObserver(String name, ConcreteSubject subject) {
        this.name = name;
        this.subject = subject;
        subject.addObserver(this);
    }


    @Override
    public void update() {
        int newState = subject.getState();
        System.out.println(name + " received an update: State is now " + newState);
    }
}

File: PHP/Behavioral/Observer/Example/ConcreteObserver.php
<?php

include 'Observer.php';
class ConcreteObserver implements Observer
{
    private $name;

    public function __construct($name) {
        $this->name = $name;
    }

    public function update($message)
    {
        echo "Observer {$this->name} received message: {$message}<br>";
    }
}